static/build-static

```diff
@@ -0,0 +1,302 @@
+TARGETPLATFORM="${2:-$CURPLATFORM}"
+
+if [ -z "$CURDIR" ] || [ -z "$TARGETPLATFORM" ]; then
+	# shellcheck disable=SC2016
+	echo 'usage: ./build-static ${CURDIR} ${TARGETPLATFORM}'
+	exit 1
+fi
+
+source "../scripts/target-platform"
+
+build_cli() {
+  [ -d "${CLI_DIR:?}/build" ] && rm -r "${CLI_DIR:?}/build"
+  (
+    cd "${CLI_DIR}"
+    set -x
+    docker buildx build \
+      --build-arg BUILDKIT_MULTI_PLATFORM=true \
+      --build-arg CGO_ENABLED="${cgo_enabled}" \
+      --build-arg DEFAULT_PRODUCT_LICENSE \
+      --build-arg PACKAGER_NAME \
+      --build-arg PLATFORM \
+      --build-arg PRODUCT \
+      --build-arg VERSION="${STATIC_VERSION}" \
+      --output ./build \
+      --platform "${TARGETPLATFORM}" \
+      --target binary .
+  )
+}
+
+build_engine() {
+  [ -d "${ENGINE_DIR:?}/bundles" ] && rm -r "${ENGINE_DIR:?}/bundles"
+  (
+    cd "${ENGINE_DIR}"
+    mkdir -p autogen # FIXME: remove when https://github.com/moby/moby/pull/43431 merged
+    set -x
+    docker buildx build \
+      --build-arg CGO_ENABLED="${cgo_enabled}" \
+      --build-arg CONTAINERD_VERSION \
+      --build-arg DEFAULT_PRODUCT_LICENSE \
+      --build-arg PACKAGER_NAME \
+      --build-arg PLATFORM \
+      --build-arg PRODUCT \
+      --build-arg RUNC_VERSION \
+      --build-arg VERSION="${STATIC_VERSION}" \
+      --output ./bundles \
+      --platform "${TARGETPLATFORM}" \
+      --target binary .
+    mkdir -p ./bundles/"${TARGETPLATFORM}"
+    cp -r ./bundles/binary-daemon/* "./bundles/${TARGETPLATFORM}/"
+  )
+}
+
+build_engine_cross() {
+  [ -d "${ENGINE_DIR:?}/bundles" ] && rm -r "${ENGINE_DIR:?}/bundles"
+  (
+    cd "${ENGINE_DIR}"
+    mkdir -p autogen # FIXME: remove when https://github.com/moby/moby/pull/43431 merged
+    set -x
+    docker buildx build \
+      --build-arg CGO_ENABLED="${cgo_enabled}" \
+      --build-arg CONTAINERD_VERSION \
+      --build-arg CROSS=true \
+      --build-arg DEFAULT_PRODUCT_LICENSE \
+      --build-arg DOCKER_CROSSPLATFORMS="${TARGETPLATFORM}" \
+      --build-arg PACKAGER_NAME \
+      --build-arg PLATFORM \
+      --build-arg PRODUCT \
+      --build-arg RUNC_VERSION \
+      --build-arg VERSION="${STATIC_VERSION}" \
+      --output ./bundles \
+      --target cross .
+    mkdir -p "./bundles/${TARGETPLATFORM}"
+    cp ./bundles/cross/"${TARGETPLATFORM}"-daemon/* "./bundles/${TARGETPLATFORM}/"
+  )
+}
+
+build_buildx() {
+  [ -d "${BUILDX_DIR:?}/bin" ] && rm -r "${BUILDX_DIR:?}/bin"
+  (
+    cd "${BUILDX_DIR}"
+    set -x
+    docker buildx build \
+      --platform "${TARGETPLATFORM}" \
+      --build-arg BUILDKIT_MULTI_PLATFORM=true \
+      --output "./bin" \
+      --target binaries .
+  )
+}
+
+build_compose() {
+  [ -d "${COMPOSE_DIR:?}/bin" ] && rm -r "${COMPOSE_DIR:?}/bin"
+  (
+    cd "${COMPOSE_DIR}"
+    set -x
+    # TODO: Add TARGETPLATFORM support on compose repo to build efficiently with --platform
+    docker buildx build \
+      --output "./bin" \
+      --target cross .
```

Comment:
For this one, I used the makefile, to prevent the chance we would be missing some build-args. Not sure we should, but looks like there's some other ones passed as well; https://github.com/docker/compose/blob/3599fc853346779910d68c52d9d3ca1b4cc25637/Makefile#L63-L68

At least we need the `GIT_TAG`

```suggestion
    make GIT_TAG="${DOCKER_COMPOSE_REF}" cross
```

Reply:
Or we could copy of course, but depends on if they add new `--build-arg`'s. Not sure if `BUILD_TAGS` is actively used for `cross`. Looks like it allows for some other options to be passed 🤔 

```suggestion
    docker buildx build \
      --build-arg BUILD_TAGS \
      --build-arg GIT_TAG="${DOCKER_COMPOSE_REF}" \
      --output "./bin" \
      --target cross .
```